lib/mongodb-filter-wrapper.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+class MongoDBFilterWrapper {
+	static get filtersAllowed() {
+		return {
+			equal: '$eq',
+			not: '$ne',
+			greater: '$gt',
+			greaterOrEqual: '$gte',
+			lesser: '$lt',
+			lesserOrEqual: '$lte',
+			in: '$in',
+			notIn: '$nin',
+			reg: '$regex',
+			all: '$all'
+		};
+	}
+
+	/**
+		* Enter door to filter process
+		* @param modelFields Object Model that will use the filter
+		* @param filters Object Filters defined as parameters
+		* @returns {{$or}|{}} Object with the filter parsed
+		*/
+	static process(model, filters) {
+		let modelFilters = {};
+
+		// Get model filters
+		if(model.constructor && model.constructor.fields)
+			modelFilters = model.constructor.fields;
+
+		// Validate filters given
+		if(typeof filters !== 'object' || (Array.isArray(filters) && filters.length === 0) || Object.entries(filters).length === 0)
+			return {};
+
+		return this.getParsedFilters(filters, modelFilters);
+	}
+
+	/**
+		* Validates filters given and loop calling to parse
+		* @returns parsedFilters Array|Object The filter parsed to use in MongoDB
+		* @param filters Object Filters defined as parameters
+		* @param modelFields Object Model that will use the filter
+		*/
+	static getParsedFilters(filters, modelFields) {
+		const orCondition = Array.isArray(filters);
+
+		let parsedFilters = orCondition ? [] : {};
+
+		// Only arrays as a filter
+		if(filters !== undefined && typeof filters === 'object' && Object.keys(filters).length > 0 && !Array.isArray(filters))
+			filters = [filters];
+
+		if(filters && filters.length > 0) {
+			// First loop over all filters given by parameters
+			filters.forEach(filterGiven => {
+				const fullyParsedFilter = this.parseObjectFilter(filterGiven, modelFields);
+				parsedFilters = (orCondition && Array.isArray(parsedFilters))
+					? [...parsedFilters, fullyParsedFilter] : Object.assign(parsedFilters, fullyParsedFilter);
+			});
+		}
+
+		if(orCondition && parsedFilters.length > 0) {
+			return {
+				$or: parsedFilters
+			};
+		}
+
+		return parsedFilters;
+	}
+
+	/**
+		* Parse the filter comparing given fields name and model fields
+		* @param filters Object Filters defined as parameters
+		* @param modelFields Object Model that will use the filter
+		* @return filterToResponse Object The filter parsed with all the values by one
+		*/
+	static parseObjectFilter(filters, modelFields) {
+		const filterToResponse = {};
+
+		for(const [field, value] of Object.entries(filters)) {
+			const fieldName = this.getFieldToCompare(field, modelFields);
+
+			const filter = (typeof filters[field] !== 'object') ? { value } : value;
```

Comment:
Podes usar `value` directamente en la condición
```suggestion
const filter = typeof value !== 'object' ? { value } : value;
```